modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/freelist/PagesList.java

```diff
@@ -613,13 +615,20 @@ protected final void put(
         assert bag == null ^ dataAddr == 0L;
 
         for (int lockAttempt = 0; ;) {
-            Stripe stripe = getPageForPut(bucket);
+            Stripe stripe = getPageForPut(bucket, bag);
```

Comment:
Here one more check like the one below is needed, because you may create a stripe inside the `getPageForPut` method

```
          if (bag != null && bag.isEmpty())
                return;
```